libs/fluid-elements/tab-group/src/lib/tab-group.ts

```diff
@@ -0,0 +1,162 @@
+
+  /** Sets the active tab on click */
+  private handleClick(event: FluidTabActivatedEvent): void {
+    if (this.activetabid != event.activeTab) {
+      this.activetabid = event.activeTab;
+      for (const tab of this.tabChildren) {
+        tab.active = tab.tabid === this.activetabid;
+      }
+      this.dispatchEvent(new FluidTabGroupActiveTabChanged(this.activetabid));
+    }
+  }
+
+  /** Sets the active tab on keydown (ArrowLeft and ArrowRight to select / Enter and Space to confirm) */
+  private handleKeyDown(event: KeyboardEvent): void {
+    // Enter Space controll (validate selection)
+    if (event.key === 'Enter' || event.key === ' ') {
+      // Set all tabs to active false
+      for (const tab of this.tabChildren) {
+        tab.active = false;
+      }
+
+      // Find the tab to be active
+      const activeTab = this.tabChildren.find((tab) => {
+        return this.activetabid === tab.tabid;
+      })!;
+
+      activeTab.active = true;
+      this.activetabid = activeTab.tabid;
+      this.dispatchEvent(new FluidTabGroupActiveTabChanged(this.activetabid));
+    }
+    // Arrow control (navigate tabs)
+    if (event.key === 'ArrowRight' || event.key === 'ArrowLeft') {
+      // Loops over to find
+      let index = this.tabChildren.findIndex(
+        (tab: FluidTab) => this.activetabid === tab.tabid,
+      );
+
+      const oldIndex = index;
+      if (event.key === 'ArrowRight') {
+        index += 1;
+      }
+      if (event.key === 'ArrowLeft') {
+        index -= 1;
+      }
+      if (index > this.tabChildren.length - 1) {
+        index = 0;
+      } else if (index < 0) {
+        index = this.tabChildren.length - 1;
+      }
+
+      this.tabChildren[index].focus();
+      this.tabChildren[index].tabindex = 0;
+      this.tabChildren[oldIndex].tabindex = -1;
+      this.activetabid = this.tabChildren[index].tabid;
+    }
+  }
+
+  private handleDisabled(disableTabEvent: FluidTabDisabledEvent): void {
+    if (this.activetabid === disableTabEvent.disableTab) {
+      const tabToEnable = this.tabChildren.find((tab) => !tab.disabled);
+      if (tabToEnable) {
+        tabToEnable.active = true;
+      }
+    }
+  }
+
+  // Handles changes in the slot. Initially sets the active item (default is first)
+  private slotchange(): void {
+    this.tabChildren = Array.from(this.querySelectorAll('fluid-tab'));
+    // Initially set the first tab to active
+    if (!this.activetabid) {
+      this.tabChildren[0].active = true;
+      this.activetabid = this.tabChildren[0].tabid;
+    } else {
+      for (const tab of this.tabChildren) {
+        tab.active = tab.tabid === this.activetabid;
+      }
+    }
+    for (const tab of this.tabChildren) {
+      tab.active = tab.tabid === this.activetabid;
+    }
+  }
+
+  /**
+   * Render function of the custom element. It is called when one of the
+   * observedProperties (annotated with @property) changes.
+   */
+  render(): TemplateResult {
+    return html`
+      <ul
```

Comment:
maybe use a `div` here. `ul` and `li` are not as valid if there are custom-elements in between.